### PR TITLE
Interative labels & related changes

### PR DESCRIPTION
**Pull request by dsehnal**
# Description

ping @corredD 

- Add `snapshotKey` and `tooltip` params to loci `LabelRepresentation`
- Update `FocusLoci` behavior to support `snapshotKey` param
  - Clicking a visual with `snapshotKey` will trigger that snapshot
- Render multiline loci label tooltips as Markdown
- `ParamDefinition.Text` updates:
  - Support `multiline` inputs
  - Support `placeholder` parameter
  - Support `disableInteractiveUpdates` to only trigger updates once the control loses focus

image

## Actions

- [x] Added description of changes to the `[Unreleased]` section of `CHANGELOG.md`
- [x] Updated headers of modified files

### CODE REVIEW

**Comment by corredD** [comment]
This looks really great and its easy to use !
Everything worked well for me except the interaction on the label. After highlighting the mouse click doesn't trigger the snapshot. Does the click need a modifier like shift or alt ?

**Comment by dsehnal** [comment]
> This looks really great and its easy to use !
> 
> Everything worked well for me except the interaction on the label. After highlighting the mouse click doesn't trigger the snapshot. Does the click need a modifier like shift or alt ?
> 
>   

Can you try giving the label background? Currently you have to click the text exactly for this to work.

**Comment by dsehnal** [comment]
@corredD I think I think what the issue is. The snapshot activation doesn't work (by design) in the "selection mode". Should we change this?

**Comment by corredD** [comment]
Ha I just realize how the selection mode work. Works great the way it is, I should have more feedback after using it more heavily. This is a really nice feature.

File: src/mol-plugin-ui/controls.tsx [hunk 1]
```diff
@@ -298,7 +298,15 @@ export class LociLabels extends PluginUIComponent<{}, { labels: ReadonlyArray<Lo
...
         return <div className='msp-highlight-info'>
-            {this.state.labels.map((e, i) => <div key={'' + i} dangerouslySetInnerHTML={{ __html: e }} />)}
+            {this.state.labels.map((e, i) => {
+                if (e.indexOf('\n') > 0) {
+                    return <div className='msp-highlight-markdown-row' key={'' + i}>
+                        <Markdown skipHtml>{e}</Markdown>
+                    </div>;
+                }
+
+                return <div className='msp-highlight-simple-row' key={'' + i} dangerouslySetInnerHTML={{ __html: e }} />;
```
Comment (arose) [inline src/mol-plugin-ui/controls.tsx:308]: We probably should use markdown everywhere where we currently use `dangerouslySetInnerHTML`. For v4?

**Review by arose — APPROVED** [review]
lgtm

**Pull request merged**